src/tools/id-mapping/adapters/idMappingConverter.ts

```diff
@@ -0,0 +1,17 @@
+import { UniParcAPIModel } from '../../../uniparc/adapters/uniParcConverter';
+import { UniProtkbAPIModel } from '../../../uniprotkb/adapters/uniProtkbConverter';
+import { UniRefAPIModel } from '../../../uniref/adapters/uniRefConverter';
+import { MappingAPIModel, MappingFlat } from '../types/idMappingSearchResults';
+
+const idMappingConverter = (data: MappingAPIModel[]): MappingFlat[] =>
+  data.map((row) => {
```

Comment:
Bit of an issue here with types. For `usePagination` to stay generic I would like to return something which extends `APIModel`. That's why I added the `R` as return type for `usePagination` but I think it has problems to match the return type.